clang/test/CIR/func-simple.cpp

```diff
@@ -57,3 +57,15 @@ bool boolfunc() { return true; }
 // CHECK:   %0 = cir.const #true
 // CHECK:   cir.return %0 : !cir.bool
 // CHECK: }
+
+float floatfunc() { return 42.42f; }
+// CHECK: cir.func @floatfunc() -> !cir.float {
+// CHECK:   %0 = cir.const #cir.fp<4.242000e+01> : !cir.float
```

Comment:
At least in LLVM-IR we are shockingly inconsistent with how we output floating point literals in IR.  So I'm not sure if we need to make the format a little less 'strict' here?  How consistent is CIR with literal representation across platforms?

Reply:
I haven't looked into how these values get printed, but I would expect the behavior to be similar to LLVM IR, which while it is as you note inconsistent is generally stable.

Reply:
> How consistent is CIR with literal representation across platforms?

We the same as MLIR LLVM dialect those (which is MLIR's own parsing/printing), we haven't tested much past AArch64 and x86_64 so not a lot of data to say much more - we haven't had issues so far though.

Reply:
As far as I understand both MLIR LLVM dialect and LLVM IR use the same format, I will update the test to just check this part `%0 = cir.const #cir.fp<4.242`